### PR TITLE
RTI-311: Prettier, TSLint and custom TSLint rules. 

### PR DESCRIPTION
**Pull request by askarby**
Introducing prettier and custom tslint rules.

### CODE REVIEW

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "arrowParens": "always",
+  "bracketSpacing": true,
+  "parser": "typescript",
+  "printWidth": 100,
```
Comment (jonasjensen-brolstark) [inline .prettierrc.json]: Is this width up for disquisition? I find it sometimes does more harm then good when splitting things after 100 chars.

File: src/app/examples/list-example/base-list.component.ts [hunk 2]
```diff
@@ -1,12 +1,17 @@
 export class BaseListComponent {
-    myItems = [
-        {title: 'Vestas Wind Systems has a very long name', subTitle: '2000 stk', amount: '5.587.218.309 DKK', detail: 225},
-        {title: 'A.P. Møller-Mærsk', subTitle: '2 stk', amount: '14.019 DKK', detail: 0},
-        {title: 'Novo Nordisk A/S B', subTitle: '18 stk', amount: '7560 DKK', detail: 171},
-        {title: 'Danske Bank A/S', subTitle: '10 stk', amount: '1560 DKK', detail: -171},
-    ];
+  myItems = [
+    {
```
Comment (jonasjensen-brolstark) [inline src/app/examples/list-example/base-list.component.ts]: Why did prettier only format one of the json objects? or was it done manually.

File: tools/tslint-rules/disallowSpecificEnvironmentImportRule.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    type: 'maintainability',
+    description: `Ensures that no imports contain specific environment.ts file (as it's handled by Angular/webpack).`,
+    options: null,
+    optionsDescription: 'Not configurable',
+    rationale: `Explicitly importing 'environment.mock.ts', 'environment.prod.ts' etc. is a no-go.`,
+    typescriptOnly: false,
+  };
+
+  static readonly FAILURE_STRING = `Do not import from specific environment file as it's handled by Angular/webpack. Remove the suffix ('.mock' or '.prod').
+*** Please note: Environment import can be simplified as: "import { environment } from '@drb-environment';"`;
```
Comment (jonasjensen-brolstark) [inline tools/tslint-rules/disallowSpecificEnvironmentImportRule.ts]: Well we don't have @drd-environment in Kirby? 😄 

We should properly change this error string.

File: .prettierrc.json (hunk 1 above)
Comment (askarby) [reply .prettierrc.json]: I just copied this from relationsbanken, and really don't know if a (consious) decision has been made for 100 characters (I assume someone has).

Prettier actually has a strong opinion of not using more than 80 characters (have a look at this link: https://github.com/prettier/prettier/issues/2375)

Maybe ask @jakobe ?

File: src/app/examples/list-example/base-list.component.ts (hunk 2 above)
Comment (askarby) [reply src/app/examples/list-example/base-list.component.ts]: I has actually formatted "all".

Prettier's philosophy is to "cram as much stuff into one line as possible", but if all properties of a JS-object can't be on a single line, it'll nicely format it (by having one property per line).

File: tools/tslint-rules/disallowSpecificEnvironmentImportRule.ts (hunk 3 above)
Comment (askarby) [reply tools/tslint-rules/disallowSpecificEnvironmentImportRule.ts]: I agree... will do!

File: src/app/examples/list-example/base-list.component.ts (hunk 2 above)
Comment (jonasjensen-brolstark) [reply src/app/examples/list-example/base-list.component.ts]: Understood, so this is the Prettier way of doing. Let follow that 😄

File: .prettierrc.json (hunk 1 above)
Comment (jonasjensen-brolstark) [reply .prettierrc.json]: So the question is more how hard we want to follow Prettier. Should we do the recommended 80 characters then?

But lets hear the opinion of @jakobe and Alexander.

File: src/main.ns.ts [hunk 4]
```diff
@@ -3,22 +3,6 @@ import { platformNativeScriptDynamic } from 'nativescript-angular/platform';
 
 import { AppModule } from './app/app.module';
 import './dev/livesync-navigation';
-const options: AppOptions = {};
```
Comment (jonasjensen-brolstark) [inline src/main.ns.ts]: Is this deletion on purpose or a merge error?

Comment (jonasjensen-brolstark) [reply src/main.ns.ts]: Doh is the change from master that deleted this. Ignore my comment.

**Pull request merged**